### PR TITLE
[APP] [ADD] Resto-APP Menu & Resto List 

### PR DESCRIPTION
**Pull request by RenanDbs**
### 🛠 Changes being made

Added task from the dev repo to the Mobile one.
Addition of the Resto list & Menu for the resto-app
Also fixed somme requested modification

### 🏎 PR checklist

1. Read the ticket
- [ ] Read the related ticket on Jira and understand the requirements for this ticket. In case anything is unclear, ask the ticket creator or ticket executor.
2. Matching PR description
- [ ] Check that the PR description on GitHub is aligned with the ticket description on Jira.
3. Review the changes on GitHub
- [ ] is the current version of the destination branch (mostly main) merged into the PR branch?
- [ ] does it fulfill the definition of done for the ticket?
- [ ] were the tests adjusted or extended for the changed behavior?
- [ ] was the documentation adjusted or extended for changed behavior?
- [ ] does it align with the team code guidelines?
- [ ] check readability of code - is it easy to understand what is happening in the code snippets?
- [ ] proper naming
- [ ] is it logical?
- [ ] are there any performance issues?
- [ ] could it be improved / simplified?
4. Execute locally / in testing environment
- [ ] Execute the code locally and check if there are any issues. Test whether the new behavior from the ticket acts as expected. Try to think of edge cases and test them.
5. Run tests
- [ ] Run the tests locally and check that all of them succeed.
6. Linting & Formatting
- [ ] Check that the changes didn’t cause any linting or formatting warnings of errors.
7. Submit Feedback
- [ ] Approve if all of the following points fulfilled:
   - all previous steps did not reveal any issues
   - there are no open comments on the PR

- [ ] otherwise, request changes with specific comments (constructive) on what to change!

### CODE REVIEW

File: restoMobile/src/components/RestaurantCard.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ const RestaurantCard = ({ info, onDelete }) => {
           resizeMode="contain"
           source={
             info.pictures[0] === 'empty.jpg'
-              ? require('../assets/logo.png')
+              ? require('/Users/duboisrenan/Guardos-mobile/restoMobile/src/assets/logo.png')
```
Comment (MarcPIster) [inline restoMobile/src/components/RestaurantCard.tsx:28]: Dont use complete paths

File: restoMobile/src/pages/AddRestaurant/AddRestaurant.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        streetNumber: streetNumber,
+        postalCode: postalCode,
+        city: city,
+        country: country,
+      },
+      description: description,
+    };
+
+    try {
+      const response = await axios.post('http://195.90.210.111:8081/api/restaurants/', restaurantData);
```
Comment (MarcPIster) [inline restoMobile/src/pages/AddRestaurant/AddRestaurant.tsx:44]: Move this to restoMobile/src/services/restoCalls.ts and use url from .env

File: restoMobile/src/pages/HomeScreen/HomeScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  website: string;
+  description: string;
+  pictures: string[];
+  hitRate?: number;
+  range: number;
+  rating: number;
+  ratingCount?: number;
+}
+
+const baseUrl = `http://195.90.210.111:8081/api/restaurants/`;
```
Comment (MarcPIster) [inline restoMobile/src/pages/HomeScreen/HomeScreen.tsx:23]: use .env

File: restoMobile/src/pages/HomeScreen/HomeScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+export const getAllResto = async () => {
```
Comment (MarcPIster) [inline restoMobile/src/pages/HomeScreen/HomeScreen.tsx:25]: move to restoMobile/src/services/restoCalls.ts and use user login token

File: restoMobile/src/pages/MenuPage/MenuPage.tsx [hunk 5]
```diff
@@ -1,107 +1,86 @@
...
-        <Text style={styles.cardTitle}>{item.dishes[0].name}</Text>
-        <Text style={styles.cardDescription}>{item.dishes[0].description}</Text>
-        <Text style={styles.cardPrice}>${item.dishes[0].price}</Text>
-      </TouchableOpacity>
-    );
-  };
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch(`http://195.90.210.111:8081/api/dishes/${restaurantName}`);
```
Comment (MarcPIster) [inline restoMobile/src/pages/MenuPage/MenuPage.tsx:34]: use restoMobile/src/services/dishsCalls.ts and use user login token

**Review by MarcPIster — CHANGES_REQUESTED** [review]

**Review by JosefineM2000 — CHANGES_REQUESTED** [review]
Please change the title of this PR according to our commit policy (specifically state the change, in a short way): https://epitech-eip.atlassian.net/wiki/spaces/EIP/pages/5963782/Commit+policy

Also, adjust the description template.

File: restoMobile/src/pages/AddRestaurant/AddRestaurant.tsx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+    if (!restaurantName || !phoneNumber || !streetName || !streetNumber || !postalCode || !city || !country || !description || !website) {
+      Alert.alert('Error', 'All fields are mandatory.');
+      return;
+    }
+    const restaurantData = {
+      name: restaurantName,
+      phonenumber: phoneNumber,
+      website: website,
+      pictures: [imageURL],
+      location: {
```
Comment (JosefineM2000) [inline restoMobile/src/pages/AddRestaurant/AddRestaurant.tsx:34]: what about opening hours as optional input?

File: restoMobile/src/pages/HomeScreen/HomeScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  name: string;
+  id: number;
+  phoneNumber: string;
+  website: string;
+  description: string;
+  pictures: string[];
+  hitRate?: number;
+  range: number;
+  rating: number;
+  ratingCount?: number;
```
Comment (JosefineM2000) [inline restoMobile/src/pages/HomeScreen/HomeScreen.tsx:22]: opening hours are missing again

File: restoMobile/src/pages/HomeScreen/HomeScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+const baseUrl = API_URL + 'restaurants/';
+
+const deleteRestaurantByName = async (restaurantName: string) => {
```
Comment (JosefineM2000) [inline restoMobile/src/pages/HomeScreen/HomeScreen.tsx:27]: Please move this to the resto call service. BE calls should only be started by a service.

File: restoMobile/src/pages/HomeScreen/HomeScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+import { useNavigation } from '@react-navigation/native';
+import Header from '../../components/Header';
+import Card from '../../components/RestaurantCard';
+import axios from 'axios';
+import styles from './HomeScreen.styles';
+import { API_URL } from '@env';
+import { getAllResto } from 'src/services/restoCalls';
+import MenuPage from '../MenuPage/MenuPage';
+
+export interface IRestaurantFrontEnd {
```
Comment (JosefineM2000) [inline restoMobile/src/pages/HomeScreen/HomeScreen.tsx:12]: interfaces should be in the models folder

File: restoMobile/src/pages/MenuPage/MenuPage.tsx [hunk 10]
```diff
@@ -1,107 +1,85 @@
...
-import { View, Text, StyleSheet, StatusBar, ImageBackground, FlatList, TouchableOpacity } from 'react-native';
-import Header from 'src/components/Header';
+import React, { useState, useEffect } from 'react';
+import { View, Text, Image, ScrollView} from 'react-native';
+import styles from './MenuPage.styles';
+import { getDishesByResto } from '../..//services/dishCalls';
 
-const MenuPage = ({ route }) => {
-  const { restaurantId, restaurantName } = route.params;
+interface Dish {
```
Comment (JosefineM2000) [inline restoMobile/src/pages/MenuPage/MenuPage.tsx:6]: please move this interface to the models folder

File: restoMobile/src/pages/MenuPage/MenuPage.tsx [hunk 11]
```diff
@@ -1,107 +1,85 @@
...
-          "name": "Homemade Pizza",
-          "description": "This is the best homemade pizza you'll ever eat. Enjoy this masterpiece of tomato and cheese",
-          "products": ["cheese", "morzerella"],
-          "price": 8.7,
-          "allergens": ["milk", "soybean"]
-        },
-      ]
-    }
-  ];
+interface DishData {
```
Comment (JosefineM2000) [inline restoMobile/src/pages/MenuPage/MenuPage.tsx:21]: this looks like its only used in this file, so you DONT have to move it to the models folder.

File: restoMobile/src/pages/MenuPage/MenuPage.tsx [hunk 12]
```diff
@@ -1,107 +1,85 @@
...
+  _id: number;
+  dishes: Dish[];
+}
 
+const MenuPage: React.FC = ({ route }) => {
+  const [dishesData, setDishesData] = useState<DishData[]>([]);
+  const [loading, setLoading] = useState(true);
+  const {restaurantId, restaurantName } = route.params;
+  console.log(restaurantName);
```
Comment (JosefineM2000) [inline restoMobile/src/pages/MenuPage/MenuPage.tsx:30]: do we still need the log or was it just for testing purposes?

File: restoMobile/src/pages/MenuPage/MenuPage.tsx [hunk 13]
```diff
@@ -1,107 +1,85 @@
...
-      </ImageBackground>
-
-      <FlatList
-      data={dishes[0].dishes}
-      renderItem={renderDishCard}
-      keyExtractor={(item) => item[0]._id.toString()}
-      style={styles.flatList}
-    />
+      {loading ? (
+        <Text>Loading...</Text>
```
Comment (JosefineM2000) [inline restoMobile/src/pages/MenuPage/MenuPage.tsx:59]: Nice!

File: visitorMobile/src/pages/MenuPage/MenuPage.tsx [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+        },
+      ]
+    }
+  ];
+
+
+  const renderDishCard = ({ item }) => {
+    console.log("item", item);
+    console.log("item.dishes", item.dishes);
+    console.log("item.dishes[0]", item.dishes[0]);
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/MenuPage/MenuPage.tsx:29]: do we still need those logs?

File: visitorMobile/src/pages/MenuPage/MenuPage.tsx [hunk 15]
```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { View, Text, StyleSheet, StatusBar, ImageBackground, FlatList, TouchableOpacity } from 'react-native';
+import Header from 'src/components/Header';
+
+const MenuPage = ({ route }) => {
+  const { restaurantId, restaurantName } = route.params;
+
+  const dishes = [
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/MenuPage/MenuPage.tsx:8]: Mh, why do we have a hardcoded dish?

File: visitorMobile/src/pages/MenuPage/MenuPage.tsx [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+        <Text style={styles.cardDescription}>{item.dishes[0].description}</Text>
+        <Text style={styles.cardPrice}>${item.dishes[0].price}</Text>
+      </TouchableOpacity>
+    );
+  };
+
+  return (
+    <View style={styles.container}>
+      <ImageBackground
+        source={{ uri: '/Users/duboisrenan/Guardos-dev/packages/restoMobile/src/assets/foodbg.png' }}
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/MenuPage/MenuPage.tsx:43]: As Marc already mentioned before, please dont use paths that only work on your device

File: visitorMobile/src/pages/MenuPage/MenuPage.tsx [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+      data={dishes[0].dishes}
+      renderItem={renderDishCard}
+      keyExtractor={(item) => item[0]._id.toString()}
+      style={styles.flatList}
+    />
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/MenuPage/MenuPage.tsx:60]: Styles should be in an extra file

File: visitorMobile/src/pages/RestaurantScreen/RestaurantScreen.tsx [hunk 18]
```diff
@@ -1,11 +1,71 @@
...
+import React, { useEffect, useState , useCallback} from 'react';
+import { View, FlatList, TouchableOpacity, Text , RefreshControl} from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+import Card from '../../components/RestaurantCard';
+import styles from './RestaurantScreen.styles'
+import axios from 'axios';
+import { getAllResto } from '../../services/restoCalls';
+import AsyncStorage from "@react-native-async-storage/async-storage";
+
+export interface IRestaurantFrontEnd {
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/RestaurantScreen/RestaurantScreen.tsx:10]: should also be in models folder.

and please add opening hours to make it complete

**Review by JosefineM2000 — CHANGES_REQUESTED** [review]
Could test it locally yet, but here already some comments from reading the code

File: restoMobile/src/pages/AddRestaurant/AddRestaurant.tsx [hunk 19]
```diff
@@ -0,0 +1,168 @@
...
+  const [streetNumber, setStreetNumber] = useState('');
+  const [postalCode, setPostalCode] = useState('');
+  const [city, setCity] = useState('');
+  const [country, setCountry] = useState('');
+  const [description, setDescription] = useState('');
+  const [website, setWebsite] = useState('');
+  const [imageURL, setImageURL] = useState('');
+
+  const handleAddRestaurant = async () => {
+    if (!restaurantName || phoneNumber || !streetName || !streetNumber || !postalCode || !city || !country || description || website) {
```
Comment (JosefineM2000) [inline restoMobile/src/pages/AddRestaurant/AddRestaurant.tsx:25]: Please remove phoneNumber, description and website from this statement - right now it will cause an error if a value is set :D

Comment (JosefineM2000) [reply restoMobile/src/pages/AddRestaurant/AddRestaurant.tsx:25]: The same checks should be done in the edit resto screen. And why cant we change the picture in the edit screen? Do you know, who was responsible for edit screen?

File: restoMobile/src/pages/AddRestaurant/AddRestaurant.tsx [hunk 20]
```diff
@@ -0,0 +1,168 @@
...
+        <View style={styles.inputPair}>
+          <TextInput
+            style={styles.input}
+            placeholder="Restaurant Name *"
+            value={restaurantName}
+            onChangeText={setRestaurantName}
+          />
+          <TextInput
+            style={styles.input}
+            placeholder="Phone Number *"
```
Comment (JosefineM2000) [inline restoMobile/src/pages/AddRestaurant/AddRestaurant.tsx:101]: Remove star that indicates required field

File: restoMobile/src/pages/AddRestaurant/AddRestaurant.tsx [hunk 21]
```diff
@@ -0,0 +1,168 @@
...
+        <View style={styles.inputPair}>
+          <TextInput
+            style={styles.input}
+            placeholder="Country *"
+            value={country}
+            onChangeText={setCountry}
+          />
+          <TextInput
+            style={styles.input}
+            placeholder="Description *"
```
Comment (JosefineM2000) [inline restoMobile/src/pages/AddRestaurant/AddRestaurant.tsx:147]: Remove star that indicates required field

File: restoMobile/src/pages/AddRestaurant/AddRestaurant.tsx [hunk 22]
```diff
@@ -0,0 +1,168 @@
...
+            value={description}
+            onChangeText={setDescription}
+            multiline
+          />
+        </View>
+
+        <TextInput
+          style={styles.input}
+          placeholder="Website *"
```
Comment (JosefineM2000) [inline restoMobile/src/pages/AddRestaurant/AddRestaurant.tsx:156]: Remove star that indicates required field

File: restoMobile/src/pages/MenuPage/MenuPage.tsx [hunk 23]
```diff
@@ -1,107 +1,70 @@
...
+      ) : (
+        <ScrollView contentContainerStyle={styles.scrollView} horizontal={false}>
+          {sortedDishes.map((dish, index) => (
+            <React.Fragment key={dish.name}>
+              {(index === 0 || sortedDishes[index - 1].category.menuGroup !== dish.category.menuGroup) && (
+                <Text style={styles.groupTitle}>{dish.category.menuGroup}</Text>
+              )}
+              <View style={styles.card}>
+                <Image source={{ uri: dish.pictures[0] }} style={styles.cardImage} />
+                <View style={styles.cardContent}>
```
Comment (JosefineM2000) [inline restoMobile/src/pages/MenuPage/MenuPage.tsx:54]: Sometimes the content extends the default dimensions of the dish card. Please make it extendable on click, so that all text is visible. Also, maybe its better to place price (and maybe even allergens) above the description, since that is more important information than description. Its fine to crop the description on limited space, but price should always be visible.

File: visitorMobile/src/pages/MenuPage/MenuPage.tsx [hunk 24]
```diff
@@ -0,0 +1,70 @@
...
+        <Text>Loading...</Text>
+      ) : (
+        <ScrollView contentContainerStyle={styles.scrollView} horizontal={false}>
+          {sortedDishes.map((dish, index) => (
+            <React.Fragment key={dish.name}>
+              {(index === 0 || sortedDishes[index - 1].category.menuGroup !== dish.category.menuGroup) && (
+                <Text style={styles.groupTitle}>{dish.category.menuGroup}</Text>
+              )}
+              <View style={styles.card}>
+                <Image source={{ uri: dish.pictures[0] }} style={styles.cardImage} />
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/MenuPage/MenuPage.tsx:53]: Sometimes the content extends the default dimensions of the dish card. Please make it extendable on click, so that all text is visible. Also, maybe its better to place price (and maybe even allergens) above the description, since that is more important information than description. Its fine to crop the description on limited space, but price should always be visible.

File: visitorMobile/src/pages/RestaurantScreen/RestaurantScreen.tsx [hunk 25]
```diff
@@ -1,11 +1,59 @@
...
+      });
+  };
+  const onRefresh = useCallback(() => {
+    setRefreshing(true);
+    updateRestoData();
+    setRefreshing(false);
+  }, []);
+
+
+  const navigateToMenu = (restaurantId: number, restaurantName: string) => {
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/RestaurantScreen/RestaurantScreen.tsx:38]: On click, the resto card should be extended. Redirecting to the menu should happen via a menu button. Also the details button is missing. Was there a reason to choose a different approach to the web version?

File: restoMobile/src/pages/HomeScreen/HomeScreen.tsx [hunk 26]
```diff
@@ -0,0 +1,84 @@
...
+    } catch (error) {
+      console.error('Error deleting restaurant:', error);
+    }
+  };
+
+  const navigateToAddRestaurant = () => {
+    navigation.navigate('AddRestaurant');
+  };
+
+  const navigateToMenu = (restaurantId: number, restaurantName: string) => {
```
Comment (JosefineM2000) [inline restoMobile/src/pages/HomeScreen/HomeScreen.tsx:46]: On click on the resto, the card should be extended. The menu should be accessable by another button next to edit and delete. - similar to the web version. Not redirect to the menu right away. Or was there a reason to choose a different approach?

File: restoMobile/src/pages/HomeScreen/HomeScreen.tsx [hunk 27]
```diff
@@ -0,0 +1,84 @@
...
+    getAllResto()
+      .then((res) => {
+        setRestoData(res);
+      })
+      .catch((error) => {
+        console.error('Error updating restaurant data:', error);
+      });
+  };
+
+  const onDelete = async (restaurantName: string) => {
```
Comment (JosefineM2000) [inline restoMobile/src/pages/HomeScreen/HomeScreen.tsx:31]: Please add a popup for verifying that the resto should be deleted to avoid actions on accident.

**Review by JosefineM2000 — CHANGES_REQUESTED** [review]
The opening hours are still missing - should be in add resto, and the resto card / resto details (both for resto and visitor version)

**Review by JosefineM2000 — APPROVED** [review]
Remaining comments will be resolved in new PR.

Additional note: currently two add resto screens exist and the wrong one is used. This also needs to be resolved in the separate PR.
